build failure on FreeBSD w/o optimization

### PureGo Version

009f4b0e473c90e938fda983e6945c5c00c4cca5

### Operating System

- [ ] Windows
- [ ] macOS
- [ ] Linux
- [x] FreeBSD
- [ ] NetBSD
- [ ] Android
- [ ] iOS

### Go Version (`go version`)

go1.22.12 (but also other versions)

### What steps will reproduce the problem?

Run the test CGO_ENABLED=0 w/o optimization

### What is the expected result?

The test passes

### What happens instead?

The test fails

https://github.com/hajimehoshi/purego/actions/runs/13751525916/job/38453081556

```
  => go test CGO_ENABLED=0 w/o optimization
  # github.com/ebitengine/purego/internal/fakecgo
  internal/fakecgo/freebsd.go:23:3: //go:cgo_export_dynamic environ only allowed in cgo-generated code
  internal/fakecgo/freebsd.go:24:3: //go:cgo_export_dynamic __progname only allowed in cgo-generated code
  FAIL	github.com/ebitengine/purego [build failed]
  FAIL	github.com/ebitengine/purego/examples/libc [build failed]
  ?   	github.com/ebitengine/purego/internal/cgo	[no test files]
  FAIL	github.com/ebitengine/purego/internal/fakecgo [build failed]
  FAIL	github.com/ebitengine/purego/internal/load [build failed]
  ?   	github.com/ebitengine/purego/internal/strings	[no test files]
  FAIL
```

### Anything else you feel useful to add?

We should add `-e` to the shebang `#!/bin/sh`.

I don't think this is happening in the stable branch, where the test is not executed in a shell script.

--- a/.github/workflows/test.yml
+++ b/.github/workflows/test.yml
@@ -7,8 +7,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        go:
-          ["1.18.x", "1.19.x", "1.20.x", "1.21.x", "1.22.x", "1.23.x", "1.24.x"]
+        go: ['1.18.x', '1.19.x', '1.20.x', '1.21.x', '1.22.x', '1.23.x', '1.24.x']
     name: Test with Go ${{ matrix.go }} on ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
     defaults:
@@ -69,8 +68,8 @@ jobs:
           # gcflags -std is necessary to make fakecgo the Cgo for
           # FreeBSD to add the symbols that libc.so depends on.
           env GOOS=freebsd GOARCH=amd64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
-          env GOOS=freebsd GOARCH=arm64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...
-
+          env GOOS=freebsd GOARCH=arm64 go build -gcflags="github.com/ebitengine/purego/internal/fakecgo=-std" -v ./...          
+          
           # Check cross-compiling NetBSD binaries.
           env GOOS=netbsd GOARCH=amd64 go build -v ./...
           env GOOS=netbsd GOARCH=arm64 go build -v ./...
@@ -116,11 +115,11 @@ jobs:
           env QEMU_LD_PREFIX=/usr/aarch64-linux-gnu qemu-aarch64 ./purego-test-nocgo -test.shuffle=on -test.v -test.count=10
           env GOOS=linux GOARCH=arm64 CGO_ENABLED=1 go test -c -o=purego-test-cgo .
           env QEMU_LD_PREFIX=/usr/aarch64-linux-gnu qemu-aarch64 ./purego-test-cgo -test.shuffle=on -test.v -test.count=10
-
+          
           echo "=> go build (plugin)"
           # Make sure that plugin buildmode works since we save the R15 register (#254)
           env GOOS=linux GOARCH=arm64 CGO_ENABLED=1 go build -buildmode=plugin ./examples/libc
-
+          
           go env -u CC
           go env -u CXX
 
@@ -133,9 +132,9 @@ jobs:
       - name: go test (Linux 386)
         if: runner.os == 'Linux'
         run: |
-          sudo apt-get install gcc-multilib
-          sudo apt-get install g++-multilib
-          env CGO_ENABLED=1 GOARCH=386 go test -shuffle=on -v -count=10 ./...
+           sudo apt-get install gcc-multilib
+           sudo apt-get install g++-multilib
+           env CGO_ENABLED=1 GOARCH=386 go test -shuffle=on -v -count=10 ./...
 
       - name: go test race (no Cgo)
         if: ${{ runner.os == 'macOS' &&  !startsWith(matrix.go, '1.18.') && !startsWith(matrix.go, '1.19.') }}
@@ -151,25 +150,16 @@ jobs:
   bsd:
     strategy:
       matrix:
-        os: ["FreeBSD"] # TODO: Add 'NetBSD' again (#304)
-        go:
-          [
-            "1.18.10",
-            "1.19.13",
-            "1.20.14",
-            "1.21.13",
-            "1.22.12",
-            "1.23.11",
-            "1.24.5",
-          ]
+        os: ['FreeBSD'] # TODO: Add 'NetBSD' again (#304)
+        go: ['1.18.10', '1.19.13', '1.20.14', '1.21.13', '1.22.12', '1.23.11', '1.24.5']
         exclude:
           # there are no prebuilt download links for these versions of Go for NetBSD
           - os: NetBSD
-            go: "1.18.10"
+            go: '1.18.10'
           - os: NetBSD
-            go: "1.19.13"
+            go: '1.19.13'
           - os: NetBSD
-            go: "1.20.14"
+            go: '1.20.14'
     name: Test with Go ${{ matrix.go }} on ${{ matrix.os }}
     runs-on: ubuntu-22.04
     defaults:

